### PR TITLE
Deprecating `Data.Fin.Dec`

### PR DESCRIPTION
**Pull request by MatthewDaggitt**
The very last of my outward-facing module level changes! 

As described in the CHANGELOG, `Data.Fin.Dec` is a non-standard module which is causing dependency cycles when I try and add a proof of the pigeonhole principle in `Data.Fin.Properties`. It also fixes the problem of the subset decidability proofs being nearly unfindable for new users.

This PR deprecates the module and moves the proofs it contains to more sensible places. The proofs have also been tidied up a little.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -77,6 +76,23 @@ Non-backwards compatible changes
...
+  subset properties easier.
+
+* The following have been moved to `Data.Fin.Properties`:
+  `decFinSubset`, `any?`, `all?`, `¬∀⟶∃¬-smallest`, `¬∀⟶∃¬`.
+
+* The following proofs have been moved to `Data.Fin.Subset.Properties`:
+  `_∈?_`, `_⊆?_`, `nonempty?`, `anySubset?` and `decLift`. The latter
+  has been renamed to `Lift?`.
+
+* The file `Data.Fin.Dec` still exists for backwards compatability
```
Comment (gallais) [inline CHANGELOG.md]: typo: compatibility

File: CHANGELOG.md [hunk 2]
```diff
@@ -869,6 +891,9 @@ Other minor additions
   _⊙?_ : Decidable P → Decidable Q → Decidable (P ⟨⊙⟩ Q)
   _⊎?_ : Decidable P → Decidable Q → Decidable (P ⟨⊎⟩ Q)
   _~?  : Decidable P → Decidable (P ~)
+
+  U?   : Decidable U
+  ∅?   : Decidable ∅
   ```
```
Comment (gallais) [inline CHANGELOG.md]: Typo: we shouldn't add anything to the already released 0.16.

Comment (gallais) [reply CHANGELOG.md]: This is something that has been bothering for quite a while: we should probably have a

File: src/Data/Fin/Properties.agda [hunk 3]
```diff
@@ -443,10 +447,64 @@ i +′ j = inject≤ (i + j) (ℕₚ.+-mono-≤ (toℕ≤pred[n] i) ℕₚ.≤-r
...
 ∀-cons z s (suc i) = s i
 
+decFinSubset : ∀ {n p q} {P : Pred (Fin n) p} {Q : Pred (Fin n) q} →
+               Decidable Q → (∀ {f} → Q f → Dec (P f)) → Dec (Q ⊆ P)
+decFinSubset {zero}  {_}     {_} _  _  = yes λ{}
+decFinSubset {suc n} {P = P} {Q} Q? P? with decFinSubset (Q? ∘ suc) P?
+... | no ¬q⟶p = no (λ q⟶p → ¬q⟶p (q⟶p))
+... | yes q⟶p with Q? zero
+...   | no ¬q₀ = yes (λ {_} → hlpr _)
+  where
```
Comment (gallais) [inline src/Data/Fin/Properties.agda]: You could write:

```yes (λ {f} → ∀-cons {P = Q U.⇒ P} (⊥-elim ∘ ¬q₀) (λ _ → q⟶p) f)` where

File: CHANGELOG.md (hunk 2 above)
Comment (MatthewDaggitt) [reply CHANGELOG.md]: Eek yes. I originally started this PR in v0.16, and hadn't noticed this was still floating there.

A directory of for past CHANGELOGs is a good idea. The single monolithic file is getting a little unwieldy.

**Pull request merged**